### PR TITLE
chaos-daemon: fix nil pointer dereference in chaos-daemon

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

When the `toda` exited unexpectedly, the process will dereference an `nil` error.

### CODE REVIEW

File: pkg/bpm/bpm.go [hunk 1]
```diff
@@ -148,8 +148,7 @@ func (m *BackgroundProcessManager) StartProcess(cmd *ManagedProcess) (*process.P
 	go func() {
 		err := cmd.Wait()
 		if err != nil {
-			err, ok := err.(*exec.ExitError)
-			if ok {
+			if err, ok := err.(*exec.ExitError); ok {
```
Comment (STRRL) [inline pkg/bpm/bpm.go:151]: I think we should change the name of `err` in `if err, ok := err.(*exec.ExitError); ok `, this re-assignment of `err` still affects the next `else` branch

Comment (STRRL) [reply pkg/bpm/bpm.go:151]: PTAL @YangKeao

Comment (YangKeao) [reply pkg/bpm/bpm.go]: Fixed.

**Review by STRRL — APPROVED** [review]
LGTM!

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Pull request merged**